### PR TITLE
Partially restore some tests for reference handles

### PR DESCRIPTION
**Pull request by csilvestrini**
There's a few edge cases that are still failing.

Also added a check that reference-mode storage keys weren't being used for reference handles

### CODE REVIEW

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 1]
```diff
@@ -431,6 +388,62 @@ open class HandleManagerTestBase {
             .isLessThan(readBack.raw!!.expirationTimestamp)
     }
 
+    @Test
+    open fun singleton_referenceLiveness() = runBlocking {
+        // Create and store an entity.
+        val writeEntityHandle = writeHandleManager.createCollectionHandle()
+        writeEntityHandle.store(entity1)
+
+        // Create a store a reference to the entity.
```
Comment (jasonwyatt) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt:397]:
```suggestion
        // Create and store a reference to the entity.
```

Comment (csilvestrini) [reply javatests/arcs/core/entity/HandleManagerTestBase.kt]: Done
